bugbug/generative_model_tool.py

```diff
@@ -20,58 +21,88 @@ def create_human_llm():
     return HumanInputLLM()
 
 
-def create_openai_llm():
+def create_openai_llm(temperature=0.2):
     from langchain_openai import ChatOpenAI
 
     return ChatOpenAI(
         model_name="gpt-4o-2024-05-13",
         api_key=get_secret("OPENAI_API_KEY"),
-        temperature=0.2,
+        temperature=temperature,
     )
 
 
-def create_azureopenai_llm():
+def create_azureopenai_llm(temperature=0.2):
     from langchain_openai import AzureChatOpenAI
 
     return AzureChatOpenAI(
         azure_endpoint=get_secret("OPENAI_API_ENDPOINT"),
         azure_deployment=get_secret("OPENAI_API_DEPLOY"),
         api_key=get_secret("OPENAI_API_KEY"),
         api_version=get_secret("OPENAI_API_VERSION"),
-        temperature=0.2,
+        temperature=temperature,
     )
 
 
-def create_anthropic_llm():
+def create_anthropic_llm(temperature=0.2):
     from langchain_anthropic import ChatAnthropic
 
     return ChatAnthropic(
         model_name="claude-3-5-sonnet-20240620",
         api_key=get_secret("ANTHROPIC_API_KEY"),
+        temperature=temperature,
     )
 
 
-def create_mistral_llm():
+def create_mistral_llm(temperature=0.2):
     from langchain_mistralai import ChatMistralAI
 
     return ChatMistralAI(
         model_name="mistral-large-latest",
         api_key=get_secret("MISTRAL_API_KEY"),
+        temperature=temperature,
     )
 
 
-AVAILABLE_LLMS = [
-    match.group(1)
-    for name in list(globals())
-    if (match := re.search(r"create_(.*?)_llm", name))
-]
+AVAILABLE_LLMS = {}
 
+for name in list(globals()):
+    match = re.search(r"create_(.*?)_llm", name)
+    if match is None:
+        continue
```

Comment:
It would still be feasible, but now that it is a dict I would need to do:
```python
AVAILABLE_LLMS = {
    name[len("create_"):-len("_llm")]: inspect.signature(globals()[name).parameters
    for name in globals()
    if re.match(r"create_(.*?)_llm", name)
]
```
which is a bit unreadable because there are too many intermediate values without a name

Reply:
I agree. We could make it even more readable by wrapping the logic in a function.

```py
AVAILABLE_LLMS = _get_llms_with_create_funcs()
```